File: comms/push-notifications/credentials/list-credentials/list-credentials.java
package com.example.usage;

import com.twilio.sdk.TwilioComms;
import com.twilio.sdk.resources.pushnotifications.credentials.requests.CredentialsListRequest;
import com.twilio.sdk.types.PushCredentialType;
import java.time.OffsetDateTime;

public class Example {
    public static void main(String[] args) {
        TwilioComms client = TwilioComms
            .builder()
            .credentials("TWILIO_ACCOUNT_SID", "TWILIO_AUTH_TOKEN")
            .build();

        client.pushNotifications().credentials().list(
            CredentialsListRequest
                .builder()
                .startDate(OffsetDateTime.parse("2024-01-15T09:30:00Z"))
                .endDate(OffsetDateTime.parse("2024-01-15T09:30:00Z"))
                .appName("appName")
                .credentialType(PushCredentialType.APN)
                .isDefault(true)
                .pageToken("pageToken")
                .pageSize(50)
                .build()
        );
    }
}

File: comms/push-notifications/credentials/list-credentials/list-credentials.cs
using Twilio.Comms;
using System.Threading.Tasks;
using Twilio.Comms.PushNotifications;
using System;
using System.Globalization;

public partial class Examples
{
    public async Task Example() {
        var client = new TwilioComms(
            accountId: "TWILIO_ACCOUNT_SID",
            authToken: "TWILIO_AUTH_TOKEN"
        );

        await client.PushNotifications.Credentials.ListAsync(
            new CredentialsListRequest {
                StartDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                EndDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                AppName = "appName",
                CredentialType = PushCredentialType.Apn,
                IsDefault = true,
                PageToken = "pageToken",
                PageSize = 50
            }
        );
    }

}
